start app as # node server.js

FabianFrank opened the pull request:
This change allows apps to be started using
$ sudo node server.js

Also they can now be deployed to nodejitsu. I think server.js is a better start script than "mojito start", for example the latter only works if mojito was installed globally, which might not be the case in a lot of production environments. server.js is more portable.

This fixes issue #126.

FabianFrank committed:
start app as # node server.js

this change allows apps to be started using
$ sudo node server.js

also they can now be deployed to nodejitsu.

FabianFrank commented:
I also made it work with heroku, just some minor changes:
https://github.com/FabianFrank/trainingMojito/compare/FabianFrank%3Amaster...heroku

http://mojito-demo.herokuapp.com/

caridy commented:
@FabianFrank this is challenging because of the variable port. In manhattan, it is manhattan the one responsible for defining the port and calling for listen on the exported server object. In dev, we let mojito to pick up the port from application.json. In nujitsu and heroku they will provide a port as well. I like your patch  to listen(process.env.PORT), maybe that's the right thing to do. I think we will need to work with manhattan to get them to use the same principle used by nujtsu and heroku.

About the appPort in application.json, I think that's just useless since you can't leverage the dimensions to have different port per env or whatever, I think we should just remove that thing and stick to the server.js definition for the listen on dev mode when process.env.PORT is not defined.

FabianFrank commented:
Thanks @caridy, let's follow up internally and I will provide updated code after we have a conclusion.

FabianFrank committed:
honoring PORT env and defaulting to 8666

8666 is more mojito-like than 80 and works by default for non-root users.

FabianFrank committed:
updating engine to 0.6.x

mojito now supports node 0.6 so there is no reason to use 0.4 any more when creating new apps. some cloud engines expect a .x version range, so going for that, because it is also shorter.

FabianFrank commented:
Since Denis and therefore Manhattan agreed to accept a server.js that calls .listen() we can change Mojito to align with what the community seems to have accepted as standard.

mojit0 commented:
Closing in favor of a more comprehensive update to Mojito startup logic: https://github.com/yahoo/mojito/pull/422

rwaldura commented:
See #422.
